source/grid/grid_in.cc

```diff
@@ -636,6 +636,78 @@ GridIn<dim, spacedim>::read_vtk(std::istream &in)
                   this->cell_data[section_name] = std::move(temp_data);
                 }
             }
+
+          // Addition of POINT_DATA
+          else if (keyword == "POINT_DATA")
+            {
+              unsigned int n_ids;
+              in >> n_ids;
+
+              AssertThrow(n_ids == vertices.size(),
+                          ExcMessage(
+                            "The VTK reader found a POINT_DATA statement "
+                            "that lists a total of " +
+                            Utilities::int_to_string(n_ids) +
+                            " cell data objects, but this needs to "
+                            "equal the number of vertices (which is " +
+                            Utilities::int_to_string(vertices.size()) + ")"));
+
+
+              in >> keyword;
+
+              // Ignore everything until we get to a SCALARS data set
+              if (keyword == "SCALARS")
+                {
+                  //
+                  std::string field_name;
+                  std::string data_type;
+                  in >> field_name;
+                  in >> data_type;
+
+                  in >> keyword;
+                  AssertThrow(
+                    keyword == "LOOKUP_TABLE",
+                    ExcMessage(
+                      "While reading VTK file, missing keyword 'LOOKUP_TABLE'."));
+
+                  in >> keyword;
+                  AssertThrow(
+                    keyword == "default",
+                    ExcMessage(
+                      "While reading VTK file, missing keyword 'default'."));
+
+                  Vector<double> temp_data(n_ids);
+                  for (unsigned int j = 0; j < n_ids; ++j)
+                    in >> temp_data[j];
+                  AssertThrow(in.fail() == false, ExcIO());
+                  this->point_data[field_name] = std::move(temp_data);
+                }
+
+              in >> keyword;
+              // Vector case
+              if (keyword == "VECTORS")
```

Comment:
Is the logic here correct? Above you have
```
              in >> keyword;

              // Ignore everything until we get to a SCALARS data set
              if (keyword == "SCALARS")
```
and here you have
```
              in >> keyword;
              // Vector case
              if (keyword == "VECTORS")
```
What happens if you have a file that has no `SCALARS` but `VECTORS`? In that case, you read one keyword, find that it is not equal to `SCALARS` (perhaps because it is equal to `VECTORS`), but then read another keyword.